lib/mix/tasks/ops.build.ex

```diff
@@ -0,0 +1,75 @@
+defmodule Mix.Tasks.Ops.Build do
+  use Mix.Task
+  alias Ops.Helper
+
+  @doc false
+  def run(args) do
+    {Helper.lookup_image_name(args && List.first(args)),
+     Ops.Shells.System.call("git", ["symbolic-ref", "--short", "-q", "HEAD"])}
+    |> put_in_console("Build service #{Mix.Project.config()[:app]}")
+    |> put_in_console("Building '#{Mix.Project.config()[:app]}' branch is ':branch:'")
+    |> put_in_console("Building '#{Mix.Project.config()[:app]}' Docker image as ':image_name:'")
+    |> write_build_info_file(Helper.settings()[:build_info_file_name])
+    |> build_docker_image()
+    |> put_in_console("Pushing ':image_name:' image into registry")
+    |> push_docker_image()
+  end
+
+  defp put_in_console({image_name, branch} = context, message) do
+    message |> String.replace(":branch:", branch) |> String.replace(":image_name:", image_name) |> Helper.puts()
+    context
+  end
+
+  defp build_docker_image({image_name, _} = context) do
+    System.find_executable("docker")
+    |> Ops.Shells.Exec.call(["build", "-f", Helper.settings()[:docker_file], "-t", image_name, "."], [{:line, 4096}])
+
+    context
+  end
+
+  defp push_docker_image({image_name, _}),
+    do: System.find_executable("docker") |> Ops.Shells.Exec.call(["push", image_name], [{:line, 4096}])
+
+  defp write_build_info_file(context, nil), do: context
+
+  defp write_build_info_file({image_name, _} = context, file) do
+    result =
+      Poison.encode!(%{
+        image: %{
+          name: image_name,
+          build_date: Timex.format!(Timex.now(), "{ANSIC}"),
+          build_author: build_author()
+        },
+        commit: %{
+          date: commit_date(),
+          message: commit_message(),
+          hash: commit_hash(),
+          author: commit_author()
+        }
+      })
+
+    file |> File.write(result)
+    context
+  end
+
+  def build_author, do: Ops.Shells.System.call("whoami", []) |> String.replace(["\n", "'"], "")
+
+  def commit_author,
+    do:
+      Ops.Shells.System.call("git", ["--no-pager", "show", "-s", "--format='%an <%ae>'"])
+      |> String.replace(["\n", "'"], "")
+
+  def commit_hash,
+    do: Ops.Shells.System.call("git", ["--no-pager", "show", "-s", "--format='%h'"]) |> String.replace(["\n", "'"], "")
+
+  def commit_message,
+    do: Ops.Shells.System.call("git", ["--no-pager", "show", "-s", "--format='%s'"]) |> String.replace(["\n", "'"], "")
```

Comment:
you can encapsulate `String.replace(["\n", "'"], "")` and run only `Ops.Shell.System`